airflow/api_fastapi/views/public/connections.py

```diff
@@ -0,0 +1,47 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+from typing_extensions import Annotated
+
+from airflow.api_fastapi.db import get_session
+from airflow.api_fastapi.openapi.exceptions import create_openapi_http_exception_doc
+from airflow.models import Connection
+
+connections_router = APIRouter(tags=["CONNECTION"])
+
+
+@connections_router.delete(
+    "/connections/{connection_id}",
+    status_code=204,
+    responses=create_openapi_http_exception_doc([401, 403, 404]),
+)
+async def delete_connection(
+    *,
+    connection_id: str,
+    session: Annotated[Session, Depends(get_session)],
+):
```

Comment:
```suggestion
) -> None:
```

Not sure, but maybe we can still annotate the return type for consistency

Reply:
I have already tried adding it in every possible way such as `None`, or any empty response to maybe generalise the approach for handling the NoContent in FastAPI. :cry: 

Adding `None` is not working unfortunately due to FastAPI requirements. FastAPI got angry because even though `None` is a `None`, it tries to see if it has a Body in validation see below error log and of course, it thinks it still has a body in this case in some exceptions since we are passing to check `None`. It gives while running the API. 

Adding any empty response fixed the issue like `JSONResponse(content={})` but then the API is returning 200 rather than 204 because somehow it still thinks we are returning something. 

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/site-packages/uvicorn/_subprocess.py", line 80, in subprocess_started
    target(sockets=sockets)
  File "/usr/local/lib/python3.8/site-packages/uvicorn/server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/usr/local/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "uvloop/loop.pyx", line 1517, in uvloop.loop.Loop.run_until_complete
  File "/usr/local/lib/python3.8/site-packages/uvicorn/server.py", line 69, in serve
    await self._serve(sockets)
  File "/usr/local/lib/python3.8/site-packages/uvicorn/server.py", line 76, in _serve
    config.load()
  File "/usr/local/lib/python3.8/site-packages/uvicorn/config.py", line 434, in load
    self.loaded_app = import_from_string(self.app)
  File "/usr/local/lib/python3.8/site-packages/uvicorn/importer.py", line 19, in import_from_string
    module = importlib.import_module(module_str)
  File "/usr/local/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/airflow/api_fastapi/main.py", line 22, in <module>
    app = cached_app()
  File "/opt/airflow/airflow/api_fastapi/app.py", line 78, in cached_app
    app = create_app()
  File "/opt/airflow/airflow/api_fastapi/app.py", line 47, in create_app
    init_views(app)
  File "/opt/airflow/airflow/api_fastapi/app.py", line 67, in init_views
    from airflow.api_fastapi.views.public import public_router
  File "/opt/airflow/airflow/api_fastapi/views/public/__init__.py", line 22, in <module>
    from airflow.api_fastapi.views.public.connections import connections_router
  File "/opt/airflow/airflow/api_fastapi/views/public/connections.py", line 36, in <module>
    async def delete_connection(
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 992, in decorator
    self.add_api_route(
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 931, in add_api_route
    route = route_class(
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 507, in __init__
    assert is_body_allowed_for_status_code(
AssertionError: Status code 204 must not have a response body

```

Reply:
Got it, thanks for trying, then we can keep it like this. 👍